packages/block-editor/src/components/block-list/block-selection-button.js

```diff
@@ -242,11 +255,22 @@ function BlockSelectionButton( { clientId, rootClientId, blockElement } ) {
 	}
 
 	const blockType = getBlockType( name );
+	let parentTitle;
+	const blockInformation = useBlockDisplayInformation(
+		getBlockRootClientId( clientId )
```

Comment:
the call to `getBlockRootClientId( clientId )` should be done inside the "useSelect"  callback above. The reason for this is that if the "root" of the clientId changes, the component won't update unless it's used in useSelect.

Reply:
@youknowriad I cannot define the useBlockInformation hook inside a callback, it must be done within a function. ESLint gave me an error when I tried.

Reply:
Oh, you mean I should just get the rootClientId from clientId in the callback? Let me try to do that.

Reply:
Done. That may work, but I haven't been able to test these changes yet. Can't get the build to work.

Reply:
Build worked after npm install again. I gave it a test and it's still working great. I believe passing the rootClientId from the selector is working just fine.